### ISSUE TITLE
Supply expect_lint_free expectation

### ISSUE THREAD

**Issue by hadley** [jimhester/lintr#30]
So you could automatically add to you unit tests

**Comment by jimhester** [comment]
Do you have any thoughts on the best way for this function to report failure?  Maybe just print the lints object first and say it failed?  Example implementation.
```r
expect_lint_free <- function(...) {
  lints <- lint_package(...)
  has_lints <- length(lints) > 0
  if (has_lints) {
    print(lints)
  }
  expectation(!has_lints, "not lint free", "lint free")
}
```
Only issue I see with this approach is maybe it wouldn't work well with some of the Non-default testthat reporters, but admittedly I don't have much experience using them.
